aws-go-lambda/Makefile

```diff
@@ -0,0 +1,4 @@
+build::
+	GOOS=linux GOARCH=amd64 go build -o ./handler/handler ./handler/handler.go
+	zip -j ./handler/handler.zip ./handler/handler
+	go build -o go-lambda main.go
```

Comment:
Do we need this last line?

Reply:
In the [example](https://github.com/pulumi/examples/tree/master/aws-go-webserver#steps) I was referring to, the first step was to build go, but to be honest, I've never tried _not_ building it.

Reply:
> EDIT: Just tried it on a different example, and seems like the changes to main.go don't persist unless I build it first.

My understanding is that if you have the binary there, then it will be used, but if you don't, then we will effectively `go run main.go`.  So what you are seeing might happen if you still have the binary there from a previous `go build`?

@EvanBoyle Is that right?  And if so, is it documented yet?

Reply:
Just tested, and you're right that pulumi builds for us. I'll make this change with #554.